### ISSUE TITLE
Remove set_dec() from timer_interrupt() once core changes land

### ISSUE THREAD

**Issue by mpe** [linuxppc/issues#408]
Nick tried to optimise `timer_interrupt()` to only call `set_dec()` when necessary in torvalds/linux@0faf20a1ad16. However that caused timer interrupt storms when booting with highres timers disabled and we had to revert it in torvalds/linux@d2b9be1f4af5.

Nick identified an issue in the timer core and the fix is now in the tip tree timers/core branch as torvalds/linux@62c1256d544747b38e77ca9b5bfe3a26f9592576.

Once the timer change hits mainline (should be v5.19) we could try the optimisation again.

**Comment by mpe** [comment]
The core timer fix was merged into v5.19-rc1, so this can now be un-reverted for a future release.

**Comment by npiggin** [comment]
Patch is in powerpc next, seems to be working as expected.

**Comment by npiggin** [comment]
This is done.

**Comment by mpe** [comment]
Fixed in torvalds/linux@c84550203b3173511e8cdbe94bc2e33175ba1d72

Thanks @npiggin :crocodile:
